src/main/java/org/zaproxy/zap/extension/jwt/fuzzer/ExtensionJWTFuzzer.java

```diff
@@ -0,0 +1,144 @@
+import org.zaproxy.zap.extension.jwt.fuzzer.ui.JWTFuzzPanelViewFactory;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.view.HttpPanelManager;
+import org.zaproxy.zap.view.popup.ExtensionPopupMenuItemMessageContainer;
+
+public class ExtensionJWTFuzzer extends ExtensionAdaptor {
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+
+    private JWTFuzzerHandler jwtFuzzerHandler;
+    private JWTMessageLocationReplacerFactory jwtMessageLocationReplacerFactory;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(1);
+        dependencies.add(ExtensionFuzz.class);
+        DEPENDENCIES = Collections.unmodifiableList(dependencies);
+    }
+
+    public ExtensionJWTFuzzer() {
+        super();
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("fuzz.httpfuzzer.description");
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    @Override
+    public String getAuthor() {
+        return Constant.ZAP_TEAM;
+    }
+
+    @Override
+    public boolean supportsDb(String type) {
+        return true;
+    }
+
+    @Override
+    public void init() {
+        jwtFuzzerHandler = new JWTFuzzerHandler();
+        jwtMessageLocationReplacerFactory = new JWTMessageLocationReplacerFactory();
+        MessageLocationReplacers.getInstance()
+                .addReplacer(HttpMessage.class, jwtMessageLocationReplacerFactory);
+    }
+
+    @Override
+    public void initView(ViewDelegate view) {
+        super.initView(view);
+
+        ExtensionScript extensionScript =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+        HttpPanelManager panelManager = HttpPanelManager.getInstance();
+        panelManager.addRequestViewFactory(
+                RequestAllComponent.NAME, new JWTFuzzPanelViewFactory(null));
+        panelManager.addRequestViewFactory(
+                RequestSplitComponent.NAME, new JWTFuzzPanelViewFactory(ViewComponent.HEADER));
+        panelManager.addRequestViewFactory(
+                RequestSplitComponent.NAME, new JWTFuzzPanelViewFactory(ViewComponent.BODY));
+
+        if (extensionScript != null) {
+            jwtFuzzerHandler.addFuzzerMessageProcessorUIHandler(
+                    new FuzzerHttpMessageScriptProcessorAdapterUIHandler(extensionScript));
+        }
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        ExtensionFuzz extensionFuzz =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionFuzz.class);
+        extensionFuzz.addFuzzerHandler(jwtFuzzerHandler);
+        extensionHook
+                .getHookMenu()
+                .addPopupMenuItem(new JWTFuzzAttackPopupMenuItem(extensionFuzz, jwtFuzzerHandler));
+        extensionHook
```

Comment:
Not sure, if i should go with the new Menu Item or not. Actually if new menu item is not created then existed fuzzer logic needs to be tweaked. So for now going with New menu item. I am not sure if icon is needed or not.

Reply:
I'll have to check the PR more closely but in theory it should not need changes in the Fuzzer add-on (nor the new menu item).

Reply:
I think you are right and i have done the changes with only one menu item